### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by jablko**
What do you think of adding minitest assertions, to improve the automated tests output?

### Before

```
  1) Error:
TestZsh#test_process_completion:
RuntimeError: timeout
    test/test_go.rb:47:in `wait'
    test/test_go.rb:118:in `until'
    test/test_go.rb:2025:in `test_process_completion'
```

### After

```
  1) Failure:
TestZsh#test_process_completion [test/test_go.rb:2047]:
Expected: "kill 16358"
  Actual: "kill 1"
```

### CODE REVIEW

**Comment by jablko** [comment]
@junegunn I've rebased this change: Would you consider giving it another look?

**Comment by jablko** [comment]
@junegunn I've brought this change up to date.

File: test/test_go.rb [hunk 1]
```diff
@@ -254,164 +245,172 @@ def test_vanilla
...
 
     tmux.send_keys :BSpace
-    tmux.until { |lines| lines[-2].include?('10/10 (2/3)') }
+    tmux.until { |lines| assert lines[-2]&.start_with?('  10/10 (2/3) ') }
 
     tmux.send_keys 'A'
     tmux.until do |lines|
-      lines[1].include?('[5 6 1]/1') && lines[-2].include?('10/10 (3/3)')
+      assert_includes lines[1], ' [5 6 1]/1 '
+      assert lines[-2]&.start_with?('  10/10 (3/3) ')
```
Comment (junegunn) [inline test/test_go.rb:413]: Any reason you put an extra space at the end? e.g. `'  10/10 (3/3) '`

File: test/test_go.rb [hunk 2]
```diff
@@ -882,38 +881,44 @@ def test_history
...
-    tmux.until { |lines| lines[-1].end_with?('> 31') }
+    tmux.until { |lines| assert_equal '> 31', lines[-1] }
     tmux.send_keys 0
-    tmux.until { |lines| lines[-1].end_with?('> 310') }
+    tmux.until { |lines| assert_equal '> 310', lines[-1] }
     tmux.send_keys :Enter
-    wait { File.exist?(history_file) && File.readlines(history_file) == %w[22 33 44 310].map { |e| e + $INPUT_RECORD_SEPARATOR } }
+    wait do
+      assert_path_exists history_file
+      assert_equal %w[22 33 44 310].map { |e| e + $INPUT_RECORD_SEPARATOR }, File.readlines(history_file)
```
Comment (junegunn) [inline test/test_go.rb:914]: Let's simplify the expression

```ruby
assert_equal %w[22 33 44 310], File.readlines(history_file).map(&:chomp)
```

Comment (jablko) [reply test/test_go.rb]: :heavy_check_mark: Done.

File: test/test_go.rb (hunk 1 above)
Comment (jablko) [reply test/test_go.rb:413]: For consistency with e.g. `'  2/10 '`, which should match `2/10` but not `2/100`, which, granted, won't actually happen in this case, but seems technically more correct?

Comment (junegunn) [reply test/test_go.rb:413]: The other strings that end with `)` don't need extra space.

Comment (jablko) [reply test/test_go.rb]: :heavy_check_mark: Removed.

**Comment by junegunn** [comment]
Nicely done, thanks for all the work.

**Pull request merged**